app/frontend/src/features/auth/jail/Jail.tsx

```diff
@@ -0,0 +1,60 @@
+import { Backdrop, makeStyles } from "@material-ui/core";
+import React, { useCallback, useEffect, useState } from "react";
+import { Redirect } from "react-router-dom";
+import Alert from "../../../components/Alert";
+import CircularProgress from "../../../components/CircularProgress";
+import PageTitle from "../../../components/PageTitle";
+import TextBody from "../../../components/TextBody";
+import { JailInfoRes } from "../../../pb/jail_pb";
+import { getJailInfo } from "../../../service/jail";
+import { useAppDispatch, useTypedSelector } from "../../../store";
+import { updateJailStatus } from "./jailActions";
+import TOSSection from "./TOSSection";
+
+const useStyles = makeStyles((theme) => ({
+  bottomMargin: { marginBottom: theme.spacing(4) },
+}));
+
+export default function Jail() {
+  const classes = useStyles(makeStyles);
+
+  const dispatch = useAppDispatch();
+  const isJailed = useTypedSelector((state) => state.auth.jailed);
+  const authError = useTypedSelector((state) => state.auth.error);
+
+  const [loading, setLoading] = useState(false);
+  const [jailInfo, setJailInfo] = useState(null as null | JailInfoRes.AsObject);
+
+  useEffect(() => {
+    (async () => {
+      setLoading(true);
+      setJailInfo(await getJailInfo());
+      setLoading(false);
+      //just in case the store is stale
+      dispatch(updateJailStatus());
+    })();
+  }, [dispatch]);
+
+  const updateJailed = useCallback(async () => {
+    setLoading(true);
+    await dispatch(updateJailStatus());
+    setLoading(false);
```

Comment:
You could solve it by putting the loading state in the redux store.

Reply:
I agree with @amdwit in this case, since you are already setting the global auth loading state when `updateJailStatus` is being dispatched, so you may as well use that here